### PR TITLE
[Maps] Improving design and a11y for attribution layer settings

### PR DESCRIPTION
**Pull request by elizabetdev**
### a11y changes

- We're using a `<fieldset />` to signal that we're going to have a group of related form fields
- We removed the EuiFormRow because it was not doing anything. And then once the popover was open the label would be focused and it seemed a bug.
- The title is a legend for that `<fieldset />`.
- Added missing `aria-label` for buttons.

### Design 

- We added a EuiHorizontalRule so that we can clearly see that the attribution is a different section. 
- The attribution lives inside a panel and the edit and clear buttons are bottom right-aligned so that they seem they control what is inside that panel.

attribution-flow@2x

### CODE REVIEW

File: x-pack/plugins/maps/public/connected_components/layer_panel/layer_settings/_attribution_popover.scss [hunk 1]
```diff
@@ -1,3 +1,3 @@
 .mapAttributionPopover {
-  width: $euiSizeXXL * 12;
+  width: $euiSizeXXL * 8;
```
Comment (nreese) [inline x-pack/plugins/maps/public/connected_components/layer_panel/layer_settings/_attribution_popover.scss]: Can we keep the original size? URLs can get very long and having more horizontal space makes the text input more usable.

**Review by nreese — COMMENTED** [review]
Thanks for putting together a new design for the the attribution.

Would it be possible to remove the horizontal line and keep the columnCompressed form row? The new design makes the attribution really stand out while its no more important then any of the other rows.

File: x-pack/plugins/maps/public/connected_components/layer_panel/layer_settings/_attribution_popover.scss (hunk 1 above)
Comment (elizabetdev) [reply x-pack/plugins/maps/public/connected_components/layer_panel/layer_settings/_attribution_popover.scss]: Yes, I'll try to add some styles to make it more similar to the `columnCompressed` form row. I'll also try to remove the horizontal line.

**Comment by elizabetdev** [comment]
@nreese here's the new design updates based on your feedback:

attribution-flow-v2@2x

**Comment by nreese** [comment]
@miukimiu That looks fantastic. Thank you

**Review by nreese — APPROVED** [review]
LGTM

**Comment by elizabetdev** [comment]
@nreese can you merge it?

`Only those with write access to this repository can merge pull requests.`

**Comment by nreese** [comment]
I have merged the changes, thanks

**Pull request merged**